clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsApplicationProvider.java

```diff
@@ -0,0 +1,147 @@
+
+    return applications;
+  }
+
+  private HashMap<String, Application> populateApplicationMap(AmazonECS amazonECS) {
+    HashMap<String, Application> applicationHashMap = new HashMap<>();
+    for (String clusterArn: amazonECS.listClusters().getClusterArns()) {
+      String clusterName = inferClusterNameFromArn(clusterArn);
+
+      ListServicesResult result = amazonECS.listServices(new ListServicesRequest().withCluster(clusterName));
+      for (String serviceArn: result.getServiceArns()) {
+        inferApplicationFromServices(amazonECS, applicationHashMap, clusterName, serviceArn);
+      }
+    }
+    return applicationHashMap;
+  }
+
+  private Set<Application> transposeApplicationMapToSet(HashMap<String, Application> applicationHashMap) {
+    Set<Application> applications = new HashSet<>();
+
+    for (Map.Entry<String, Application> entry :applicationHashMap.entrySet()) {
+      applications.add(entry.getValue());
+    }
+
+    return applications;
+  }
+
+  private HashMap<String, Application> inferApplicationFromServices(AmazonECS amazonECS,
+                                                                    HashMap<String, Application> applicationHashMap,
+                                                                    String clusterName,
+                                                                    String serviceArn) {
+    DescribeServicesResult describeResult = amazonECS.describeServices(new DescribeServicesRequest().withServices(serviceArn).withCluster(clusterName));
+
+    for (Service service: describeResult.getServices()) {
+      if (service.getServiceName().contains(SERVICE_DELIMITER)) {
+        String appName = inferAppNameFromServiceName(service.getServiceName());
+
+        HashMap<String, String> attributes = new HashMap<>();  // After POC we'll figure exactly what info we want to put in here
+        attributes.put("iamRole", service.getRoleArn());
+        attributes.put("taskDefinition", service.getTaskDefinition());
+        attributes.put("desiredCount", String.valueOf(service.getDesiredCount()));
+
+        HashMap<String, Set<String>> clusterNames = new HashMap<>();
+        clusterNames.put(appName, Sets.newHashSet(service.getServiceName()));
+
+        EcsApplication application = new EcsApplication(appName, attributes, clusterNames);
+
+        if (!applicationHashMap.containsKey(appName)) {
+          applicationHashMap.put(appName, application);
+        } else {
+          applicationHashMap.get(appName).getAttributes().putAll(application.getAttributes());
+          applicationHashMap.get(appName).getClusterNames().get(appName).add(service.getServiceName());
+        }
+      }
+    }
+
+    return applicationHashMap;
+  }
+
+  private String inferAppNameFromServiceName(String serviceName) {
+    return serviceName.split(SERVICE_DELIMITER)[0];
+  }
+
+  private String inferClusterNameFromArn(String clusterArn) {
```

Comment:
Add javadoc about the expected format of the clusterArn

Reply:
An ARN is an ARN

Reply:
ARNs from different services have different formats (http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html#arns-syntax). I doubt that having a quick reference to the format of the string that this method is trying to parse is a useless comment.